packages/front-end/components/DecisionCriteria/DecisionCriteriaModal.tsx

```diff
@@ -0,0 +1,638 @@
+              onChange={(e) => form.setValue("name", e.target.value)}
+              required
+              disabled={disabled}
+            />
+          </div>
+          <Text weight="bold" size="2">
+            Description
+          </Text>
+          <div className="form-group">
+            <textarea
+              className="form-control"
+              placeholder="(optional)"
+              value={form.watch("description")}
+              onChange={(e) => form.setValue("description", e.target.value)}
+              rows={2}
+              disabled={disabled}
+            />
+          </div>
+        </Flex>
+
+        <Text weight="bold" size="2">
+          Rules
+        </Text>
+        <Text size="2">
+          Rules are evaluated in order. If a rule matches, an action is
+          recommended and no further rules are evaluated.
+        </Text>
+
+        {form.watch("rules").map((rule, ruleIndex) => (
+          <Flex
+            key={rule.id}
+            className="appbox mb-1 mt-1"
+            direction="column"
+            gap="2"
+            p="2"
+          >
+            <Flex justify="between" align="center" mb="1">
+              <Text weight="bold" size="2">
+                Rule {ruleIndex + 1}
+              </Text>
+              {ruleIndex > 0 && !disabled && (
+                <IconButton
+                  variant="ghost"
+                  color="red"
+                  size="1"
+                  onClick={() => removeRule(rule.id)}
+                >
+                  <PiTrash />
+                </IconButton>
+              )}
+            </Flex>
+
+            {rule.conditions.map((condition, conditionIndex) => (
+              <Flex key={condition.id} direction="column">
+                <Flex align="center" gap="4" width="100%">
+                  <Box width="40px">
+                    <Text size="2" color={"gray"}>
+                      {conditionIndex === 0 ? "If" : "and"}
+                    </Text>
+                  </Box>
+
+                  <Box style={{ flex: 1 }}>
+                    <Select
+                      size={selectSize}
+                      value={condition.match}
+                      setValue={(value) =>
+                        updateCondition(rule.id, condition.id, "match", value)
+                      }
+                      disabled={disabled}
+                    >
+                      {MATCH_OPTIONS.map((option) => (
+                        <SelectItem key={option.value} value={option.value}>
+                          {option.label}
+                        </SelectItem>
+                      ))}
+                    </Select>
+                  </Box>
+
+                  <Box style={{ flex: 1 }}>
+                    <Select
+                      size={selectSize}
+                      value={condition.metrics}
+                      setValue={(value) => {
+                        if (value === "guardrails") {
```

Comment:
question: Is it intentional we are calling updateCondition twice for `guardrails` or should we have an else?